src/win/process.c

```diff
@@ -1069,27 +1067,62 @@ int uv_spawn(uv_loop_t* loop,
     process_flags |= CREATE_SUSPENDED;
   }
 
-  if (!CreateProcessW(application_path,
-                     arguments,
-                     NULL,
-                     NULL,
-                     1,
-                     process_flags,
-                     env,
-                     cwd,
-                     &startup,
-                     &info)) {
-    /* CreateProcessW failed. */
-    err = GetLastError();
-    goto done;
+  if (options->flags & UV_PROCESS_WINDOWS_RUNAS_ADMIN) {
+    shellexecuteinfo.cbSize = sizeof(SHELLEXECUTEINFOW);
+    shellexecuteinfo.fMask = fMask | SEE_MASK_WAITFORINPUTIDLE  | SEE_MASK_NOCLOSEPROCESS; // for shellexecuteinfo.hProcess
+    shellexecuteinfo.lpVerb = L"runas";
+    shellexecuteinfo.lpFile = application_path;
+    shellexecuteinfo.lpParameters = arguments;
+    shellexecuteinfo.nShow = startup.wShowWindow;
+    if (!ShellExecuteExW(&shellexecuteinfo)) {
+        err = GetLastError();
+        goto done;
+    }
+    hProc = shellexecuteinfo.hProcess;
```

Comment:
The documentation states this sometimes returns NULL instead of the hProcess. How should libuv deal with that?

Reply:
> Even if fMask is set to SEE_MASK_NOCLOSEPROCESS, hProcess will be NULL if no process was launched

Is no process actually have been launched considered an error?

Reply:
On Unix systems you'd get an error like ENOENT.

Reply:
done